CombinatorialGames/Game/Specific/Domineering.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2019 Kim Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kim Morrison, Violeta Hernández Palacios
+-/
+import CombinatorialGames.Game.Concrete
+import Mathlib.Algebra.Group.Units.Equiv
+import Mathlib.Algebra.Order.Group.Nat
+
+/-!
+# Domineering as a combinatorial game.
+
+We define the game of Domineering, played on a chessboard of arbitrary shape
+(possibly even disconnected).
+Left moves by placing a domino vertically, while Right moves by placing a domino horizontally.
+
+This is only a fragment of a full development;
+in order to successfully analyse positions we would need some more theorems.
+Most importantly, we need a general statement that allows us to discard irrelevant moves.
+Specifically to domineering, we need the fact that
+disjoint parts of the chessboard give sums of games.
+-/
+
+namespace PGame
+
+open Function
+
+private theorem pred_fst_ne (m : ℤ × ℤ) : (m.1 - 1, m.2) ≠ m := by aesop
+private theorem pred_snd_ne (m : ℤ × ℤ) : (m.1, m.2 - 1) ≠ m := by aesop
```

Comment:
```suggestion
-- Note: `aesop` can't prove pred_*_ne when the inequality is reversed
private theorem pred_fst_ne (m : ℤ × ℤ) : (m.1 - 1, m.2) ≠ m := by aesop
private theorem pred_snd_ne (m : ℤ × ℤ) : (m.1, m.2 - 1) ≠ m := by aesop
```

Reply:
(I also would like to be more sparing of the use of `aesop`, and this would probably be the only place where I would suggest using `have ...; simp` instead of `aesop`, but since this is a private theorem, this makes sense :+1:).

Reply:
Just realized I only ended up using these theorems once... I think I can just inline them and replace them by a short term-mode proof.

Reply:
I do think the other places in the file where I used `aesop` are quite appropriate uses of it, e.g. the theorem
```
(ofDomineering (toDomineering (((ofDomineering b).erase m).erase (m.1 - 1, m.2)))).card + 2 = (ofDomineering b).card
```
is "structurally trivial" but would take a couple of lines of code to prove manually.